src/client/Client.cc

```diff
@@ -3934,6 +3934,7 @@ void Client::remove_cap(Cap *cap, bool queue_release)
   }
   size_t n = in->caps.erase(mds);
   assert(n == 1);
+  delete cap; // remove myself from session's cap list
```

Comment:
`cap` is destructed via `in->caps.erase`. This is a recent change: a3be080d03a76870fd28f2bc7a609fd8a6c3aa8f

Reply:
IOW, there should not be a leak...

Reply:
yeah！ didn't notice the in->caps has changed, thanks for your explanation. there are no problem at all, sorry, shall close this one